contributors/design-proposals/standardization-of-development-process.md

```diff
@@ -0,0 +1,282 @@
+may be completed through a few rounds of review in order to achieve alignment.
+
+### Git and GitHub Implementation
+
+Practically an RFC would be implemented as a pull request to a central repository
+with the following example structure
+
+```
+├── 0000-rfc-template.md
+├── CODEOWNERS
+├── index.md
+├── sig-architecture
+├── sig-network
+│   └── kube-dns
+├── sig-node
+│   └── kubelet
+├── sig-release
+├── sig-storage
+├── unsorted-to-be-used-by-newcomers-only
+└── wg-resource-management
+```
+
+where each SIG or working group is given a top level directory with components
+maintained by the SIG listed in sub directories. For newcomers to the community
+an `unsorted-to-be-used-by-newcomers-only` directory may be used before an RFC
+can be properly routed to a SIG although hopefully if discussion for a potential
+RFC begins on the mailing lists proper routing information will be provided to
+the RFC author. Additionally a top level index of RFCs may be helpful for people
+looking for a complete list of RFCs. There should be basic CI to ensure that a
+`index.md` remains up to date.
+
+Ideally no work would begin on an RFC before it has been approved by the
+responsible SIG or working group. In order to help combat the explosion of GitHub
+issues against `kubernetes/kubernetes` it is further proposed for each chartered
+SIG to create a top level repository within the `kubernetes` GitHub organization
+where implementation issues would be filed and linked to in the RFC.
+
+### RFC Editor Role
+
+Taking a cue from the [Python PEP process][], I believe that a group of RFC editors
+will be required to make this process successful; the job of an RFC editor is
+likely very similar to the [PEP editor responsibilities][] and will hopefully
+provide another opportunity for people who do not write code daily to contribute
+to Kubernetes.
+
+[Python PEP process]: https://www.python.org/dev/peps/pep-0001/
+[PEP editor responsibilities]: https://www.python.org/dev/peps/pep-0001/#pep-editor-responsibilities-workflow
+
+### Important Metrics
+
+I believe the primary metrics which would signal the success or failure of the RFC
+process would be
+
+- distribution of RFC merge times
+- RFC rejection rate
+- distribution of time between RFC acceptance and implementation issue creation
+- PRs referencing an RFC merged per week
+
+### Customization of the RFC Process
+
+Given the general independence of SIGs I believe it will be important for SIGs to
+decide if they would like to opt out of the RFC process entirely or if they would
+like to modify the RFC process to better serve their needs. For example I can
+imagine RFCs targeting SIG Architecture or SIG API Machinery would likely require
+a section on backwards compatibility and migration strategies. I believe the best
+place for a SIG to describe its development process would be in its charter.
+
+### Prior Art
+
+The RFC process as proposed was essentially stolen from the [Rust RFC process] which
```

Comment:
I think quite a bit of the process is missing here. Do these other processes include:

* How reviewers and approvers are assigned
* Approval decision process
* State machine for expected procedure
* Strawman schedule/deadline/timeframe for each stage
* Communication/notification mechanisms (@sebgoa commented on this also: https://groups.google.com/d/msg/kubernetes-dev/65A-3ULYPB0/xJ1vF-EbBwAJ)
* Review meetings / escalation
* Decision on where development should occur (org/repo/branch)

Reply:
Agree -- but we don't have to have all of that decided out the gate.  But perhaps we should have a TODO section with that list.

Reply:
I moved most of those questions to the `unresolved questions` section to be addressed in future proposals